### DISCUSSION TITLE
Access pimcore admin session data

### DISCUSSION THREAD

**Discussion by klickparkdominik** [pimcore/pimcore#15973] [Q&A / Support] [answered]
I would like to know whether a user is admin or not in a custom route.

The idea is to access to current request data.
There should be a session available with the logged-in user information.

This is an example, the idea I was thinking of:

```
namespace AppBundle\Controller;

...
use Pimcore\Controller\FrontendController;
use Symfony\Component\HttpFoundation\Request;
...

class MyController extends FrontendController
{
    /**
     * @Route("my/route")
     */
    public function view(Request $request)
    {
        // Get active session
        $session = $request->getSession();

        var_dump($session->all());

        $bag = $session->getBag('pimcore_admin');
        $user = $bag->get('user');

        if (!$user || !$user->getAdmin()) {
            throw new NotFoundHttpException('Not found');
        }
....
```

The problem is that `$session->all()` is always empty.

Is the implementation wrong?
Or maybe some configuration in security.yaml missing?

In older versions of Pimcore this was no problem.
I solved it with the following code, which is no longer working in version 11.
So basically I need to rewrite this:

```
        $user = \Pimcore\Tool\Session::getReadonly()->get("user");
        if (!$user || !$user->getAdmin()) {
            throw new NotFoundHttpException('Not found');
        }
```

I would appreciate any small tips.

**Comment by klickparkdominik** [comment]
To add some more context to this.
The function `getReadonly()` has been removed in 11.

**Version 10.6**
https://github.com/pimcore/pimcore/blob/dd81ef4c666b18c254333867a60f6ed455025076/lib/Tool/Session.php#L219

**Version 11**
https://github.com/pimcore/pimcore/blob/11.x/lib/Tool/Session.php

So basically the question is, how could I solve this "the new way".

**Comment by Maksonis** [comment, marked as answer]
Maybe this way:
`$user = Pimcore\Tool\Authentication::authenticateSession();`
`$isAdmin = $user->isAdmin()`
?

**Reply by klickparkdominik** [reply]
Hi @Maksonis,

thank you for your reply. 

Even though this looks promising, `$user` is `null` when executing your lines.
I called `https://my.pimcore.tld/my/route` which executes the controller.

I will try and dig a bit deeper into the `Authentication` class.

This is the full example with your code:


```
namespace AppBundle\Controller;

...
use Pimcore\Controller\FrontendController;
use Symfony\Component\HttpFoundation\Request;
...

class MyController extends FrontendController
{
    /**
     * @Route("my/route")
     */
    public function view(Request $request)
    {
       $user = \Pimcore\Tool\Authentication::authenticateSession($request);
       var_dump($user);

       $isAdmin = $user->isAdmin();
....
```

**Reply by klickparkdominik** [reply]
I have to admit, this example is working now.

I am not 100% sure why it did not work in the first try. 
I was logged in and the session must have existed.

Whatever. Thank you so much for your answer, it is very helpful.

**Reply by klickparkdominik** [reply]
The reason why it wasn't working in the first try is the following:

I had to add `- { path: ^/my/route roles: ROLE_PIMCORE_USER }` to security.yaml to make it working.
